### ISSUE TITLE
[Idea] Raspberry Pi hat PCB

### ISSUE THREAD

**Issue by probonopd** [probonopd/MiniDexed#88]
[klonk2905](https://www.reddit.com/user/klonk2905/) wrote over at [Reddit](https://www.reddit.com/r/synthdiy/comments/syup5a/comment/hy2twxk/?utm_source=reddit&utm_medium=web2x&context=3):

> Hey, I'm willing to collaborate on the hardware part.
>
> A "hat" with decent power supply, quality DAC, and dedicated sliders/controls will be top notch at some point, and that is something I can help with.

Let's collect the ideas and requirements here.

* [Alphanumeric 1602 LCD Display](https://www.berrybase.de/en/sensors-modules/displays/alphanumeric-displays/alphanumerisches-lcd-16x2-gr-252-n/gelb)
*  [KY-040 rotary encoder](https://www.berrybase.de/en/components/passive-components/potentiometer/rotary-encoder/drehregler/rotary-encoder-mit-breakoutboard-ohne-gewinde-und-mutter)
* Footprint to solder in [GY PCM5102 DAC module](https://github.com/dwhinham/mt32-pi/wiki/GY-PCM5102-DAC-module) ([schematic](https://twitter.com/todbot/status/1528074704455471104/photo/3)) and the [CJMCU 5102 DAC module](https://github.com/dwhinham/mt32-pi/wiki/CJMCU-5102-DAC-module)
* Serial RX/TX DuPont header pins (for 3.3V serial speaking MIDI, e.g., to connect to a computer using a Raspberry Pi Pico)
* Footprint for optional Serial MIDI circuit (for "real" MIDI devices)
* Needs to work with all Rasperry Pi models
* (Deprioritized) Optional circuitry for using analog control voltages? (for Eurorack use)

Basically a https://github.com/gmcn42/clumsyMIDI but with 1602 LCD and rotary encoder.

**Comment by D4p0up** [comment]
(Klonk 2905 Here) Thanks for letting me in. 

Great suggestions, a few philosophical questions first : 
- what is the interaction philosophy you would expect from such hardware? Should it mimic the "frugality" of the original - i.e. one 16x2 display and a couple of buttons" - or do we push further (rotary encoders, ribbons, tactile displays, etc...)?
- What is the form factor expected? Matchbox ultra integrated PIZero Synth, desktop machine, 19' Rack?

Should the direction be settled, I could propose some quick mockups on Openscad to match the interaction/mechanical/PCB/RPi cinematics with the form factor/concept.

**Comment by probonopd** [comment]
Hello @D4p0up, welcome.

Yes, I'd say let's mimic the "frugality" of the original. In fact, our UI is even way more simplistic than the original (16x2 display with just 1 rotary encoder and 1 button so far). If we go down the path of making hardware, we should possibly add just a few more buttons to allow for future expanadability and/or UI shortcuts. Possibly we should think about making it compatible with [mt32-pi](https://github.com/dwhinham/mt32-pi) as well.

> What is the form factor expected?

Minimalistic, inexpensive, using jellybean parts. People shold be able to make them at home.

My first idea was to make a Raspberry Pi Hat that sits on top of a Pi with the least extra space needed and in a cost-effective way. 

Actually there is already one for the [mt32-pi](https://github.com/dwhinham/mt32-pi#%EF%B8%8F-project-status). Just missing "our" rotary encoder and button, and a different display.

But that makes for a relatively dull box (like my 3D printed prototype on the [README](https://github.com/probonopd/MiniDexed#minidexed-) page). People might also want to stick them into Euroracks, possibly 19" racks. To allow for those form factors, one would probably need to use a ribbon cable from the Pi to the MiniDexed PCB.

What are others' thoughts?

**Comment by dcoredump** [comment]
Hi @D4p0up  and welcome!

a decend hat with audio codec and DIN MIDI I/O would be very nice. The hardware audio outputs should be on two 6.3mm jacks and perhaps also with an automaticly  mono switch when using only the left output. This would make MiniDexed perfectly fitting for a connection to mixing console as well as for a live gig (only connected mono to a monitor box).

I think the minimalistic display and one encoder is enough for fast switching sounds and some parameters. For programming an external MIDI editor can be used. Or someone writes a UI for a HDMI touch display like for [minisynth](https://github.com/rsta2/minisynth).

**Comment by bengarland** [comment]
Would ClumsyMIDI work with MiniDexed?

**Comment by probonopd** [comment]
The 0.91" OLED display is currently not supported by MiniDexed, only 16x2 alphanumeric displays without i2c at the moment. Although this [might](https://github.com/luisfcorreia/MiniDexed/commit/ada781ee334fdb25d99a04d04cfed13ad60b308d) change in the future if @luisfcorreia succeeds.

**Comment by diyelectromusic** [comment]
Re: ClumsyMIDI.

If you leave off the I2C display, Clumsy MIDI works fine with MiniDexed for serial MIDI and the PCM1502 I2S audio output if you configure it for I2S audio.  You have to add the rotary encoded and 1602 LCD as extras as before of course, but GPIO wise it all works out fine.

I have a few notes on this here: https://diyelectromusic.wordpress.com/2022/04/28/bare-metal-raspberry-pi-minidexed-dx7/

Kevin

**Comment by JohnConnor84** [comment]
Please a 4x4 Gpio Membrane keypad menu implementation would be so awesome! pleaaase! My idea is indeed a eurorack format or a 2U 19inch rack version.

**Comment by probonopd** [comment]
@JohnConnor84 you have now posted the idea "4x4 Gpio Membrane keypad" in 3 different places.
The best place to start a discussion around this would be [here](https://github.com/probonopd/MiniDexed/discussions/new?category=ideas).

**Comment by probonopd** [comment]
@diyelectromusic has made one out of perfboard: :+1: 

![image](https://user-images.githubusercontent.com/2480569/166631880-b5b6081a-d94d-47f7-b022-69f3f4314df4.png)

https://diyelectromusic.wordpress.com/2022/05/03/bare-metal-raspberry-pi-minidexed-dx7-part-2/

**Comment by probonopd** [comment]
https://github.com/probonopd/MiniDexed/issues/163#issuecomment-1120297634

**Comment by probonopd** [comment]
Note to self: https://aisler.net/, including parts

**Comment by stevelittlefish** [comment]
I'm also interested in this and considering designing a PCB.  Some ideas:

- I'd much rather ditch the rotary encoder module as it makes it more difficult with a PCB.  I would put the encoder straight on the PCB with pull up resistors and a simple RC filter for debouncing (I've done this in the past with a 10k resistor and 10nF capacitor for some eurorack modules).
- I would use SMD parts for the small bits a pieces and get them pre-assembled.  They would be small and not really cost enough to care about.
- Ideally I'd put the DAC chip on the PCB and just replicate one of those modules, but it's probably too time consuming and would increase the cost (those modules are cheap).
- I would like either RCA connectors or dual 6.35mm TS jacks, but that might make it too big.
- I'd also like a DIN midi input.  I could probably get the optocoupler presoldered.  If a full sized DIN wouldn't fit then TRS mini jack could work.
- The LCD screen I've been using for testing has the connector in a different place and it has 2 rows of pins instead of 1.  Rubbish for breadboarding but fine for the PCB.  Ideally I'd make it compatible with both layouts.
- In addition to the encoder I would put a single push button to function as a back button (long press for home) and use the encoder button only to select the current menu item.
- Trimmer potentiometer somewhere for contrast would be nice. Probably with optional additional holes to just add a resistor.

I can do the schematic and PCB layout and would share all of the files.  You could then order PCBs directly from JLCPCB (including pick and place).  I could possibly send a few boards out too.

For an enclosure someone could design a 3d-printable box, or we could do it semi enclosed with a front and back panel (made out of PCB material) and screwed through the corners using machine screws and stand offs.

I think the hardest part would probably be designing it so that all of the user interface components fit in a nice layout and are the correct distance to the front panel.  This would be easier if we forgo the enclosure.

**Comment by diyelectromusic** [comment]
> @diyelectromusic has made one out of perfboard: 👍

By the way, I now have a PCB version too... See discussion here: https://github.com/probonopd/MiniDexed/discussions/335

**Comment by probonopd** [comment]
![image](https://user-images.githubusercontent.com/2480569/188329084-88abeedf-f91e-45fa-8ac0-f56e06e9834e.png)

https://diyelectromusic.wordpress.com/2022/08/16/minidexed-raspberry-pi-io-board-part-3/

Great work @diyelectromusic :+1:
